### ISSUE TITLE
Button

### ISSUE THREAD

**Issue by aVileBroker** [Headstorm/hs-angular-ui#7]
AC:
* Consistent styling across browsers
* 0 A11y violations
* > 90% code coverage
* API:
* * text
the text shown in the button
no default, optional
string
* * type
the button type, can be "default", "primary", or "destructive"
default is "default", optional
string
* * isProcessing
if true, shows a spinner to the left of the button text
default is false, optional
boolean
* * isLoading
if true, replaces the button text with a skeleton version
default is false, optional
boolean
* * iconPrefix
the name of the material design icon to be shown to the left of the button
no default, optional
string
* * iconSuffix
the name of the material design icon to be shown to the right of the button
no default, optional
string
